cli.js

```diff
@@ -1,84 +1,132 @@
-    'Outputs logs in "tail -f" fashion')
-  .action(namedActionHandler(logHandler));
-
-commander
-  .command('create')
-  .description('Create a function from template')
-  .option('-f, --function <name>',
-    'Name of the function to generate. If omitted, a name will be chosen at random')
-  .action(actionWrapper(createHandler));
+yargs
+  .option('path', {
+    alias: 'p',
+    describe: 'Use directory dir. "create" will create this directory if needed.',
+    type: 'string',
+    global: true,
+  })
+  .usage(
+`Binaris command line interface
 
-commander
-  .command('deploy <function>')
-  .description('Deploys a function to the cloud')
-  .action(namedActionHandler(deployHandler));
+Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
+  )
+  .command('create [function] [options]', 'Create a function from template', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 create [function] [options]')
+      .positional('function', {
+        describe: 'Name of the function to generate',
+        type: 'string',
+      })
+      .option('random', {
+        alias: 'r', describe: 'Generate a random name for your function', type: 'boolean',
+      });
+  }, async (argv) => {
+    if (argv.random) {
+      // eslint-disable-next-line no-param-reassign
+      argv.function = makeNameValid(moniker.choose());
+    }
+    await handleCommand(argv, createHandler);
+  })
+  .command('deploy <function> [options]', 'Deploys a function to the cloud', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 deploy <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to deploy',
+        type: 'string',
+      });
+  }, async (argv) => {
+    await handleCommand(argv, deployHandler);
+  })
+  .command('remove <function> [options]', 'Remove a previously deployed function', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 remove <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to remove',
+        type: 'string',
+      });
+  }, async (argv) => {
+    await handleCommand(argv, removeHandler);
+  })
+  .command('invoke <function> [options]', 'Invoke a Binaris function', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 invoke <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to invoke',
+        type: 'string',
+      })
+      .option('json', {
+        alias: 'j',
+        describe: 'Path to file containing JSON data',
+        type: 'string',
+      })
+      .option('data', {
+        alias: 'd',
+        describe: 'Data to send with invocation',
+        type: 'string',
+      });
+  }, async (argv) => {
+    await handleCommand(argv, invokeHandler);
+  })
+  .command('logs <function> [options]', 'Print the logs of a function', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 logs <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to retrieve logs for',
```

Comment:
> Name of the function to retrieve logs for
A bit verbose. It's clear enough if it's just `Function name`. Obviously, from the context, it is clear that this fetches the logs for it...

Reply:
Agreed.